cpp/roaring.hh

```diff
@@ -462,18 +502,26 @@ public:
   }
 
   bool operator<(const type_of_iterator &o) {
+    if (!i.has_value) return false;
+    if (!o.i.has_value) return true;
```

Comment:
not checking has_value caused the end of the iterator to compare equal to the last valid value of the iterator for every function